data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/EventMetadataImpl.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.model.event;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.time.Instant;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A Basic implementation of {@link EventMetadata} interfaces utilizing an immutable map for attributes.
+ *
+ * @since 1.2
+ */
+public class EventMetadataImpl implements EventMetadata {
```

Comment:
Is there another name we can use here? I try to avoid `Impl` naming in public classes, and even more so for model classes.

Reply:
I'd suggest `DefaultEventMetadata` or something like that.